### PR TITLE
Implement the pure-virtual functions in the Waitable class.

### PR DESCRIPTION
**Pull request by clalancette**
This is the counterpart to https://github.com/ros2/rclcpp/pull/2593

### CODE REVIEW

**Review by alsora — APPROVED** [review]

File: test_rclcpp/test/test_waitable.cpp [hunk 1]
```diff
@@ -87,6 +87,11 @@ class WaitableWithTimer : public rclcpp::Waitable
     execute_promise_.set_value(RCL_RET_OK == ret);
   }
 
+  void set_on_ready_callback(std::function<void(size_t, int)>) override {}
+  void clear_on_ready_callback() override {}
+
+  std::shared_ptr<void> take_data_by_entity_id(size_t) override {return nullptr;}
```
Comment (ahcorde) [inline test_rclcpp/test/test_waitable.cpp:93]: include `cstddef`

Comment (clalancette) [reply test_rclcpp/test/test_waitable.cpp:105]: Fixed in 817af53

**Comment by clalancette** [comment]
CI for this is in https://github.com/ros2/rclcpp/pull/2593#issuecomment-2263366528 .  I'm going to go ahead and merge this one in.

**Review by fujitatomoya — APPROVED** [review]
lgtm

**Pull request merged**